### DISCUSSION TITLE
How to use theme colors (e.g. green-500) as the default primary color?

### DISCUSSION THREAD

**Discussion by kdb13** [twbs/bootstrap#36792] [Q&A]
Bootstrap 5 has various theme colors like `gray-500`, `green-500`, etc. If we want to use these colors anywhere else after importing `_variables.scss` it works perfectly fine.

But suppose I want to change the `$primary` theme variable to a value from these built-in theme colors, like this:
```
$primary: $green-500;    // This doesn't work

@import "shoppr/scss/bootstrap/bootstrap";
```

It says the variable **green-500** is not found and I understand as we need to import `_variables.scss` first. But if we do that then we loose the functionality of overriding the `$primary` variable.

**Comment by TravelTimN** [comment]
@KdB13 

Try this:

```scss
/* import the original Bootstrap files first */
@import "functions";
@import "variables";

/* next, set whatever customization you want */
$primary: $green-500;
$secondary: $pink-500;

$theme-colors: (
  "primary": $primary,
  "secondary": $secondary
);

/* finally, import Bootstrap to accept your changes */
@import "bootstrap";
```

**Reply by kdb13** [reply]
@TravelTimN I tried your solution and it works for only the colors which we specified in the map like in your case they are `primary` and `secondary`.

All the other colors are removed due to the fact that we are creating a new map and the default map will be ignored by Bootstrap due to the `!default` field.

In the [Bootstrap docs](https://getbootstrap.com/docs/5.2/customize/sass/#modify-map), they mention it just like you did.

Do we need to change the position of this map in our `custom.scss` file so that it only modifies some colors but keeps the others (like **danger**) intact?

**Comment by kdb13** [comment]
Just found the solution from this [site](https://blog.assistancy.be/blog/bootstrap-5-override-theme-colors-sass).

We need to redefine all the remaining default colors to get them along with our custom ones:

```
/* import the original Bootstrap files first */
@import "functions";
@import "variables";

// First override some or all individual color variables
$primary: $green-500;
$secondary: $pink-500;

// Then add them to your custom theme-colors map, together with any additional colors you might need
$theme-colors: (
  "primary":   $primary,
  "secondary": $secondary,
  "success":    $success,
  "info":       $info,
  "warning":    $warning,
  "danger":     $danger,
  "light":      $light,
  "dark":       $dark
);

/* finally, import Bootstrap to accept your changes */
@import "bootstrap";
```

**Reply by TravelTimN** [reply]
Excellent! Glad you figured out the remaining color addition. Hopefully you're able to get that sorted for your project now!
